include/drivers/modem/hl7800.h

```diff
@@ -500,6 +500,16 @@ void mdm_hl7800_register_gpio6_callback(void (*func)(int state));
  */
 void mdm_hl7800_register_cts_callback(void (*func)(int state));
 
+/**
+ * @brief Set the bands available for the LTE connection
+ *
+ * @param bands Band bitmap in hexadecimal format without the 0x prefix.
+ * Leading 0's for the value can be ommited.
+ *
+ * @return int32_t negative errno, 0 on success
+ */
+int32_t mdm_hl7800_set_bands(const char *bands);
```

Comment:
Why not take `int` as argument and just use snprintf to print it as hex with leading zeros?

Reply:
I wanted to make the API easier for the user.
So like the AT command, you can set the bands without the leading zeros and the user doesn't need to know the number of leading zeros.

Reply:
If you do it similar to

https://github.com/LairdCP/zephyr/blob/PR/hl7800_band/drivers/modem/hl7800.c#L5255:#L5258

where this function takes number as argument and `snprintk` to hex form with leading zeros, I think the user also doesn't need to know the length of leading zeros?

```c
int32_t mdm_hl7800_set_bands(0x1234567890123456789012);

or 

int32_t mdm_hl7800_set_bands(0x12, 0x123456, 0x12345678901234);
```

Reply:
This would mean I need to make the function take a variable amount of args. Also, args would not be a string. I do not want this approach. Simply passing the bands as a string is more straight forward.